protocol/communities/manager.go

```diff
@@ -2870,8 +2870,7 @@ func (m *Manager) checkChannelPermissions(viewOnlyPermissions []*CommunityTokenP
 	}
 	response.ViewOnlyPermissions.Permissions = viewOnlyPermissionsResponse.Permissions
 
-	if (hasViewOnlyPermissions && !viewOnlyPermissionsResponse.Satisfied) ||
-		(hasViewOnlyPermissions && !hasViewAndPostPermissions) {
+	if hasViewOnlyPermissions && !hasViewAndPostPermissions {
```

Comment:
I believe the whole function logic could be simplified to:
```golang
  response.ViewAndPostPermissions.Satisfied = hasViewAndPostPermissions && viewAndPostPermissionsResponse.Satisfied
  response.ViewOnlyPermissions.Satisfied = viewAndPostPermissionsResponse.Satisfied || (hasViewOnlyPermissions && viewOnlyPermissionsResponse.Satisfied)
```

Reply:
I think `viewAndPostPermissionsResponse.Satisfied` is `true` if `hasViewAndPostPermissions` is `false`, you know what I mean? So we'd need a more extensive condition. 

There is surely a way to simplify, but I don't feel like messing with it :joy:

Reply:
I agree that this function needs some love applied, it's quite easy to make a mistake in such logic operations.
But I understand your caution 😄